cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -494,13 +495,15 @@ def _find_duplicate_params(param_list):
     return duplicated_params
 
 
-def _find_overlapping_paths(paths_list):
+def _find_overlapping_paths(shared_paths_list, local_paths_list):
     overlapping_paths = []
-    if paths_list:
-        for path in paths_list:
-            is_overlapping = any(x for x in paths_list if x != path and x.startswith(path + "/"))
+    if shared_paths_list:
+        for path1, path2 in list(combinations(shared_paths_list, 2)) + list(
```

Comment:
If we need to keep this, let's add a comment describing what we are doing with this for with combinations and product calls.

Reply:
Done